File: solutions/c/collatz-conjecture/2/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
    int count = 0;
    if (start == 1) {
        return count;
    }

    while (start > 1)
    {
        if (start % 2 == 0) {
            start = start / 2;
        } else {
            start = (start * 3) + 1;
        }
        if (start == 1) {
            count += 1;
            return count;
        } else {
            count += 1;
        }
    }
    return -1;
}

File: solutions/typescript/collatz-conjecture/1/collatz-conjecture.ts
export function steps(count: number): number {
    if (!Number.isInteger(count) || count <= 0) {
        throw new Error('Only positive integers are allowed')
    }
    let steps = 0

    while (count > 1) {
        count = count % 2 === 0 ? count / 2 : count * 3 + 1
        steps++
    }

    return steps
}
